examples/timeseries/timeseries_weather_forecasting.py

```diff
@@ -0,0 +1,418 @@
+
+    x_data = [dataframe.index.hour, dataframe.index.month, dataframe.index.year]
+    x_labels = ["Hour of Day", "Month", "Year"]
+
+    for i in range(3):
+        selected_ax = axes[i]
+        sns.lineplot(x=x_data[i], y=dataframe[selected_feature], ax=selected_ax)
+        selected_ax.set(xlabel=x_labels[i], ylabel=titles[idx])
+    plt.tight_layout()
+    plt.show()
+    return
+
+
+# plot pressure
+show_time_based_visualizations(df, 0)
+# plot temperature
+show_time_based_visualizations(df, 1)
+# plot specific humidity
+show_time_based_visualizations(df, 8)
+
+"""
+## Data Preprocessing
+
+Here we are picking 300000 data points for training. The observation is recorded every
+10 mins, that means 6 times in an hour. Thus STEP is equal to 6. We are tracking data
+from past 720 timestamps(i.e. 720/6=120 hours) and this will be used to predict the
+temperature after 72 timestamps(i.e. 76/6=12 hours).
+
+Raw data is normalized using a z score formula. Since every feature has values with
+varying ranges, normalization is done to confine the values in a range of [0,1] before
+training a neural network.
+It is done by subtracting the mean and dividing by the standard deviation of each feature
+"""
+
+train_split = 300000
+past = 720
+future = 72
+step = 6
+learning_rate = 0.005
+validation_steps = 50
+
+batch_size = 256
+buffer_size = 10000
+evaluation_interval = 200
+epochs = 10
+
+
+def normalize(data, train_split):
+    data_mean = data[:train_split].mean(axis=0)
+    data_std = data[:train_split].std(axis=0)
+    return (data - data_mean) / data_std
+
+
+"""
+We will be using selecting few parameters from the dataset to avoid complete memory(RAM)
+usage. Moreover, we can see from the correlation heatmap, few parameters like
+Relative Humidity and Specific Humidity are redundant.
+"""
+
+feature_idx = [0, 1, 5, 7, 8, 10, 11]
+print("The selected parameters are:", ", ".join([titles[i] for i in feature_idx]))
+selected_features = [feature_keys[i] for i in feature_idx]
+features = df[selected_features]
+features.index = df[date_time_key]
+features.head()
+
+features = normalize(features.values, train_split)
+features = pd.DataFrame(features)
+features.head()
+
+train_data = features.loc[0 : train_split - 1]
+val_data = features.loc[train_split:]
+
+# training dataset
+start = past + future
+end = start + train_split
+
+x_train = train_data[[i for i in range(7)]].values
+y_train = features.iloc[start:end][[1]]
+
+sequence_length = int(past / step)
+
+dataset_train = tf.keras.preprocessing.timeseries_dataset_from_array(
+    x_train,
+    y_train,
+    sequence_length=sequence_length,
+    sampling_rate=step,
+    batch_size=batch_size,
+)
+
+
+# validation dataset
+start = train_split + past + future
+end = len(val_data) - past - future
+
+x_val = val_data.iloc[:end][[i for i in range(7)]].values
+y_val = features.iloc[start:][[1]]
+
+dataset_val = tf.keras.preprocessing.timeseries_dataset_from_array(
```

Comment:
Please add a text paragraph explaining what `timeseries_dataset_from_array` does and why use it.